src/software/backend/output/wifi/communication/protobuf_primitive_visitor.cpp

```diff
@@ -0,0 +1,163 @@
+#include "software/backend/output/wifi/communication/protobuf_primitive_visitor.h"
+
+#include "shared/constants.h"
+#include "shared/proto/primitive.pb.h"
+#include "software/ai/primitive/all_primitives.h"
+
+PrimitiveMsg ProtobufPrimitiveVisitor::getPrimitiveMsg()
```

Comment:
We should unit test this class.

Reply:
Will port over the `mrf_primitive_visitor` tests on the final PR